addons/sale_stock/models/product_template.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    @api.onchange('type')
+    def _onchange_type(self):
```

Comment:
Same as above I would add a docstring to explain the behavior and why we set those options.